### PR TITLE
[psc-ide] Return qualifier from explicit/hiding imports

### PR DESCRIPTION
**Pull request by nwolverson**

### CODE REVIEW

File: src/Language/PureScript/Ide/Types.hs [hunk 1]
```diff
@@ -149,16 +149,16 @@ instance ToJSON ModuleImport where
     object $ [ "module" .= mn
              , "importType" .= ("implicit" :: Text)
              ] ++ fmap (\x -> "qualifier" .= x) (maybeToList qualifier)
-  toJSON (ModuleImport mn (P.Explicit refs) _) =
-    object [ "module" .= mn
+  toJSON (ModuleImport mn (P.Explicit refs) qualifier) =
+    object $ [ "module" .= mn
            , "importType" .= ("explicit" :: Text)
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/Types.hs]: Keep the alignment of `[` and `,` please

File: src/Language/PureScript/Ide/Types.hs [hunk 2]
```diff
@@ -149,16 +149,16 @@ instance ToJSON ModuleImport where
...
            , "identifiers" .= (identifierFromDeclarationRef <$> refs)
-           ]
-  toJSON (ModuleImport mn (P.Hiding refs) _) =
-    object [ "module" .= mn
+           ] ++ fmap (\x -> "qualifier" .= x) (maybeToList qualifier)
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/Types.hs]: The `Protolude` is imported here so just `map` is actually the `Functor` version.

File: src/Language/PureScript/Ide/Types.hs [hunk 3]
```diff
@@ -149,16 +149,16 @@ instance ToJSON ModuleImport where
...
+  toJSON (ModuleImport mn (P.Hiding refs) qualifier) =
+    object $ [ "module" .= mn
            , "importType" .= ("hiding" :: Text)
            , "identifiers" .= (identifierFromDeclarationRef <$> refs)
-           ]
+           ] ++ fmap (\x -> "qualifier" .= x) (maybeToList qualifier)
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/Types.hs]: Same two comments as above

**Comment by kRITZCREEK** [comment]
Oh yeah and it would be nice if you could add the two fields to `psc-ide-server/PROTOCOL.md`, I'm trying to keep it up-to-date.

File: src/Language/PureScript/Ide/Types.hs (hunk 2 above)
Comment (nwolverson) [inline src/Language/PureScript/Ide/Types.hs]: Hmm, this was ambiguous due to the non-qualified import of `Data.Map.Lazy`. Had no idea Haskell had both `import qualified ... as ...` and `import ... as ...` forms...

Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/Types.hs]: Haskell's namespacing is way too complex...

**Pull request merged**